stock_location_tank/models/stock_picking_type.py

```diff
@@ -0,0 +1,25 @@
+# Copyright NuoBiT Solutions - Frank Cespedes <fcespedes@nuobit.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class PickingType(models.Model):
+    _inherit = "stock.picking.type"
+
+    tank_filling = fields.Boolean()
+
+    @api.constrains("tank_filling")
+    def _check_tank_filling(self):
+        for rec in self:
+            if (
+                rec.env["stock.picking.type"].search_count(
```

Comment:
You forgot to filter by `warehouse_id`. Also check if it'll work if the user has multiple companies selected.